python_modules/dagster/dagster/_cli/code_server.py

```diff
@@ -144,6 +146,13 @@ def code_server_cli():
     help="How long to wait for code to load or reload before timing out. Defaults to no timeout.",
     envvar="DAGSTER_CODE_SERVER_STARTUP_TIMEOUT",
 )
+@click.option(
+    "--server-heartbeat-timeout",
+    type=click.INT,
+    required=False,
+    default=DEFAULT_SERVER_HEARTBEAT_TIMEOUT,
+    help="How long to wait for a heartbeat from the caller before timing out. Defaults to 30 seconds.",
+)
```

Comment:
this should default to off (like dagster api grpc does) - we still want to support the existing use case of users running their own grpc servers, not dagster dev managing it, in which case there will not be a heartbeat - changing the default behavior here would break that use case